core/src/Kokkos_View.hpp

```diff
@@ -694,7 +694,8 @@ class View : public Impl::BasicViewFromTraits<DataType, Properties...>::type {
 // FIXME_NVCC: nvcc 12.2 and 12.3 view these as ambiguous even though they have
 // exclusive requirements clauses. 12.6 Also has some issues though it manifests
 // differently
-#if defined(KOKKOS_ENABLE_CUDA) || defined(KOKKOS_COMPILER_NVHPC)
+#if defined(KOKKOS_COMPILER_NVCC) || defined(KOKKOS_COMPILER_NVHPC) || \
+    (defined(KOKKOS_COMPILER_CLANG) && defined(KOKKOS_ENABLE_CUDA))
```

Comment:
I assume `clang` to be a better compiler when it comes to `requires` clauses so I would remove the guard here (or rename the workaround macro)
```suggestion
#if defined(KOKKOS_COMPILER_NVCC) || defined(KOKKOS_COMPILER_NVHPC)
```

Reply:
Mind if we delay this till after 5.0? I want this PR to just do the minimal change to fix a build that showed up. I made a conscious decision in the original PR to include clang because there were some failures with it.